.github/workflows/codeql-analysis.yml

```diff
@@ -7,34 +7,182 @@ on:
     # The branches below must be a subset of the branches above
     branches: [master]
   schedule:
-    - cron: '0 1 * * 4'
+    # At 01:37 every Thursday. Details in https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule
+    - cron: '37 1 * * 4'
 
 jobs:
+  create-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get languages from repo
+        id: set-matrix
+        uses: advanced-security/set-codeql-language-matrix@v1
+        with:
+          access-token: ${{ github.token }}
+          endpoint: ${{ github.event.repository.languages_url }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.languages }}
+
   analyze:
     name: Analyze
-    if: github.repository == 'photoview/photoview'
-    runs-on: ubuntu-20.04
-
-    # strategy:
-    #   fail-fast: false
-    #   matrix:
-        # Override automatic language detection by changing the below list
-        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
-        # language: ['go', 'javascript']
-        # Learn more...
-        # https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#overriding-automatic-language-detection
-
+    needs: create-matrix
+    if: ${{ needs.create-matrix.outputs.matrix != '[]' && github.repository == 'photoview/photoview' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ ( matrix.language == 'go' && 'golang:1.23-bookworm' ) || ( matrix.language == 'javascript' && 'node:18' ) || 'null' }}
+      volumes:
+        - /opt/hostedtoolcache:/opt/hostedtoolcache
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ${{ fromJSON(needs.create-matrix.outputs.matrix) }}
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+
+      - name: Install GO build dependencies
+        if: ${{ matrix.language == 'go' }}
+        working-directory: ./api
+        shell: bash
+        run: |
```

Comment:
No need to set `working-directory` before. With that, `run` could be:

```
export TARGETPLATFORM="linux/amd64"

git config --global --add safe.directory `pwd`
# Comment out the last line with 'apt-get install -y reflex sqlite3'
sed -i '$ s/^/#/' ../scripts/install_build_dependencies.sh

source ./scripts/set_compiler_env.sh
./scripts/install_build_dependencies.sh
apt-get install -y file
echo "<<< Dependencies installed >>>"

go env
go mod download
# Patch go-face
sed -i 's/-march=native//g' ${GOPATH}/pkg/mod/github.com/!kagami/go-face*/face.go
# Build dependencies that use CGO
go install github.com/mattn/go-sqlite3 github.com/Kagami/go-face
```

Reply:
I thought that these GO commands should be executed from the repo's ./api folder, as it was originally implemented in the testing workflow (before the switch to the container). Are you sure that they will be executed correctly from the repo's root?

Reply:
Those scripts are installing dependencies. We could change to `api` after setting env.